doc/rst/technotes/gpu.rst

```diff
@@ -117,26 +116,31 @@ Setup and Compilation
 ---------------------
 
 To enable GPU support set the environment variable ``CHPL_LOCALE_MODEL=gpu``
-before building Chapel. Chapel's build system will automatically try and deduce
-where your installation of CUDA exists. If the build system fails to do this,
-or you would like to use a different CUDA installation, you can set
-``CHPL_CUDA_PATH`` environment variable to the CUDA installation root.
+before building Chapel. 
+
+Chapel's build system will automatically try and deduce what type of GPU you
+have and where your installation of relevant runtime (e.g. CUDA or ROCM) are.
+If the type of GPU is not detected you may set ``CHPL_GPU_CODEGEN`` manually to
+either ``cuda`` (for NVIDIA GPUs) or ``rocm`` (for AMD GPUs). If the relevant
+runtime path is not automatically detected (or you would like to use a
+different installation) you may set ``CHPL_CUDA_PATH`` and/or
+``CHPL_ROCM_PATH``.
+
+You may also have to set the ``CHPL_GPU_ARCH`` environment variable.  When
+``CHPL_GPU_CODEGEN`` is set to ``cuda`` by default we target compute capability
+6.0 (``--cuda-gpu-arch=sm_60`` in clang).  When ``CHPL_GPU_CODEGEN`` is set to
+``rocm`` we assign ``CHPL_GPU_ARCH`` to ``gfx906`` by default.  You may modify
+this by changing  ``CHPL_GPU_ARCH`` or by passing it to ``chpl`` via
+``--gpu-arch``.
 
 We also suggest setting ``CHPL_RT_NUM_THREADS_PER_LOCALE=1`` (this is necessary
 if using CUDA 10).
 
-To compile a program simply execute ``chpl`` as normal. By default the generated
-code will target compute capability 6.0 (specifically by passing
-``--cuda-gpu-arch=sm_60`` when invoking clang). If you would like to target a
-different compute capability (necessary for example, when targeting Tesla K20
-GPUs) you can pass ``--gpu-arch`` to ``chpl`` and specify a different value
-there.  This may also be set using the ``CHPL_GPU_ARCH`` environment variable.
-
-If you would like to view debugging information you can pass ``--verbose`` to
-your generated executable. This output will show the invocation of CUDA kernel
-calls along with various other interactions with the GPU such as memory
-operations.  You may also use the :mod:`GPUDiagnostics` module to gather
-similar information.
+To compile a program simply execute ``chpl`` as normal.  If you would like to
+view debugging information you can pass ``--verbose`` to your generated
```

Comment:
We don't have this anymore and replaced it with `--debugGpu`, right? In any case, I don't think it is something we should advertise going forwards, and while here, you can drop that sentence.

I'm still hoping to do a reorganization pass over this technote, I can do it then, too.

Reply:
Yeah, I might as well update it now.  I'll change this paragraph to:

```
To compile a program simply execute ``chpl`` as normal.  To ensure that
a loop is executing on the GPU you can use the operations in
the :mod:`GPUDiagnostics` module or use the :proc:`~GPU.assertOnGpu()`
proc from the  :mod:`GPU` module.
````

Reply:
Looks good. Just make sure that you use `GpuDiagnostics` after you fix the merge conflicts. This re-write might show up as non-conflict, I never know.